hazelcast-jclouds/src/test/java/com/hazelcast/jclouds/ComputeServiceBuilderInvalidConfigTest.java

```
@@ -0,0 +1,139 @@
+                                      "<property name=\"role-name\">test</property>";
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void test_invalid_config_when_tag_keys_and_values_have_different_sizes() throws Exception {
+
+        String propertiesUnderTest = "<property name=\"provider\">aws-ec2</property>" +
+                                     "<property name=\"identity\">test</property>" +
+                                     "<property name=\"credential\">test</property>" +
+                                     "<property name=\"tag-keys\">tag1,tag2,tag3</property>" +
+                                     "<property name=\"tag-values\">tag1,tag2</property>";
+
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = RuntimeException.class)
+      public void test_invalid_config_when_port_config_exceed_max_value() throws Exception {
+
+        String propertiesUnderTest = "<property name=\"provider\">aws-ec2</property>" +
+                "<property name=\"identity\">test</property>" +
+                "<property name=\"credential\">test</property>" +
+                "<property name=\"hz-port\">78000</property>";
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = RuntimeException.class)
```

`InvalidConfigurationException` should be thrown instead of RuntimeException.. That is our fail fast standard.